src/main/java/org/janelia/saalfeldlab/n5/codec/N5Codecs.java

```diff
@@ -0,0 +1,268 @@
+				final int numBytes = header.numElements() * dataCodec.bytesPerElement();
+				final ReadData decompressed = compression.decode(ReadData.from(in), numBytes);
+				dataCodec.deserialize(decompressed, data);
+				return dataBlockFactory.createDataBlock(header.blockSize(), gridPosition, data);
+			}
+		}
+	}
+
+	/**
+	 * DataBlockCodec for N5 data type STRING
+	 */
+	static class StringDataBlockCodec implements DataBlockCodec<String[]> {
+
+		private static final Charset ENCODING = StandardCharsets.UTF_8;
+		private static final String NULLCHAR = "\0";
+
+		private final Compression compression;
+
+		StringDataBlockCodec(final Compression compression) {
+			this.compression = compression;
+		}
+
+		@Override
+		public ReadData encode(final DataBlock<String[]> dataBlock) throws IOException {
+			return ReadData.from(out -> {
+				final String flattenedArray = String.join(NULLCHAR, dataBlock.getData()) + NULLCHAR;
+				final byte[] serializedData = flattenedArray.getBytes(ENCODING);
+				new ChunkHeader(dataBlock.getSize(), serializedData.length).writeTo(out);
+				compression.encode(ReadData.from(serializedData)).writeTo(out);
+				out.flush();
+			});
+		}
+
+		@Override
+		public DataBlock<String[]> decode(final ReadData readData, final long[] gridPosition) throws IOException {
+			try(final InputStream in = readData.inputStream()) {
+				final ChunkHeader header = ChunkHeader.readFrom(in, MODE_DEFAULT, MODE_VARLENGTH);
+				final ReadData decompressed = compression.decode(ReadData.from(in), header.numElements());
+				final byte[] serializedData = decompressed.allBytes();
+				final String rawChars = new String(serializedData, ENCODING);
+				final String[] actualData = rawChars.split(NULLCHAR);
+				return new StringDataBlock(header.blockSize(), gridPosition, actualData);
+			}
+		}
+	}
+
+	/**
+	 * DataBlockCodec for N5 data type OBJECT
+	 */
+	static class ObjectDataBlockCodec implements DataBlockCodec<byte[]> {
+
+		private final Compression compression;
+
+		ObjectDataBlockCodec(final Compression compression) {
+			this.compression = compression;
+		}
+
+		@Override
+		public ReadData encode(final DataBlock<byte[]> dataBlock) throws IOException {
+			return ReadData.from(out -> {
+				new ChunkHeader(null, dataBlock.getNumElements()).writeTo(out);
+				compression.encode(ReadData.from(dataBlock.getData())).writeTo(out);
+				out.flush();
+			});
+		}
+
+		@Override
+		public DataBlock<byte[]> decode(final ReadData readData, final long[] gridPosition) throws IOException {
+			try(final InputStream in = readData.inputStream()) {
+				final ChunkHeader header = ChunkHeader.readFrom(in, MODE_OBJECT);
+				final byte[] data = new byte[header.numElements()];
+				final ReadData decompressed = compression.decode(ReadData.from(in), data.length);
+				new DataInputStream(decompressed.inputStream()).readFully(data);
+				return new ByteArrayDataBlock(null, gridPosition, data);
+			}
+		}
+	}
+
+	static class ChunkHeader {
```

Comment:
I'd call this `BlockHeader` since everything in N5 uses "block" not "chunk"

Reply:
done